seismic/receiver_fn/rf_process_io.py

```diff
@@ -23,12 +25,16 @@ def async_write(rfstream_queue, outfile_name, max_buffered=100):
     :type outfile_name: str or Path
     :param max_buffered: Maximum number of RFStreams to buffer before flushing to file, defaults to 100
     :type max_buffered: int, optional
+    :param metadata: Metadata string to write to root attribute
+    :type metadata: str, optional
     """
     buffered_streams = []
     logger = logging.getLogger(__name__)
     logger.setLevel(logging.INFO)
     logger.info("Starting async write thread")
-    first_write = True
+    with h5py.File(outfile_name, mode='w') as h5f:
```

Comment:
When writing RFs to file, now writes the job configuration dictionary to the root metadata node for easier traceability.